src/features/hotspots/root/HotspotsList.tsx

```diff
@@ -123,22 +291,49 @@ const HotspotsList = ({
         )}
       </Box>
     )
-  }, [visibleHotspots, visible, order, hasOfflineHotspot, t])
+  }, [
+    visibleHotspots,
+    followedValidators.length,
+    locationBlocked,
+    fleetModeEnabled,
+    gatewaySortOrder,
+    validators.length,
+    hasOfflineHotspot,
+    t,
+  ])
 
   const renderItem = useCallback(
     ({ item }) => {
-      return (
-        <HotspotListItem
-          onPress={handlePress}
-          hotspot={item}
-          showCarot
-          loading={loadingRewards}
-          totalReward={rewards[item.address]}
-          hidden={hiddenAddresses?.includes(item.address)}
-        />
-      )
+      if (isHotspot(item)) {
+        return (
+          <HotspotListItem
+            onPress={handlePress}
+            gateway={item}
+            showCarot
+            loading={loadingHotspotRewards}
+            totalReward={hotspotRewards[item.address]}
+            hidden={hiddenAddresses?.includes(item.address)}
+          />
+        )
+      }
+      if (isValidator(item)) {
+        return (
+          <ElectedValidatorItem
```

Comment:
Maybe we refactor this to `ValidatorListItem`?

Reply:
that sounds good